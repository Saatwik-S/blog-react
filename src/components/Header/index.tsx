import React from 'react'

export default function Header (): JSX.Element {
  return (
        <header className='paddings'>
            <div className='title'>
                <p>
                    The Artifact
                    <br />
                    <em> Culture & Art Blog </em>
                </p>
            </div>
            <nav>
                <a href='index.html'>Blog</a>
                <a href='about.html'>About</a>
                <a href='contact.html'>Contact</a>
            </nav>
        </header>
  )
}
