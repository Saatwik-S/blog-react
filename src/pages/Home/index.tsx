import React, { useEffect } from 'react'
import Header from '../../components/Header'
import Footer from '../../components/Footer'
import Card, { type CardProps } from '../../components/Card'
import { makeRequest } from '../../utils/makeRequest'
import { GET_BLOG_POSTS } from '../../constants'
export default function Home (): JSX.Element {
  const [data, setData] = React.useState<CardProps[] | null>(null)
  const [error, setError] = React.useState<Error | null>(null)
  useEffect(() => {
    makeRequest(GET_BLOG_POSTS()).then(setData).catch((error) => {
      console.log(error.message)

      setError(error)
    })
  }, [])
  if (error != null) {
    return <div>{error.message}</div>
  }

  return (
        <div className='body'>
            <Header />
            <div className='main-body paddings'>
               {(data != null)
                 ? data?.sort((a, b) => (a.title.localeCompare(b.title))).map((item, index) => (
                    <Card key={index} {...item} />
                 ))
                 : <div>Loading...</div>}
            </div>
            <Footer />
        </div>
  )
}
