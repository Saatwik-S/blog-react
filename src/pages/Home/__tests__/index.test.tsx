import { render, screen, waitFor } from '@testing-library/react'
import Home from '..'
import React from 'react'
import Header from '../../../components/Header'

describe('Home Page', () => {
  // it('should check if all card components are there are not', async () => {
  //   const { asFragment } = render(<Home />)
  //   expect(screen.getAllByTestId('box')
  //   ).toHaveLength(6)
  //   await waitFor(() => {
  //     screen.getByText('How painting outdoors can transform your ')
  //   })
  //   expect(asFragment).toMatchSnapshot()
  // })
})
