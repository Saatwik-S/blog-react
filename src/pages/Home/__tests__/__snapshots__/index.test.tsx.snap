// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should check if all card components are there are not 1`] = `[Function]`;
